packages/element/src/create-interpolate-element.js

```diff
@@ -161,7 +145,8 @@ const createInterpolateElement = ( interpolatedString, conversionMap ) => {
  */
 const isValidConversionMap = ( conversionMap ) => {
 	return typeof conversionMap === 'object' &&
-		typeof conversionMap.length === 'undefined';
+		Object.values( conversionMap )
+			.some( ( element ) => isValidElement( element ) );
```

Comment:
did you mean `Object.values( … ).every( … )` instead of `.some( … )`?

It's okay to be strict and require that we don't pass in extra invalid elements. But right now as long as we have _one_ value that's a `WPElement` we are allowing any number of other invalid values

Reply:
I intentionally changed it to `some` because this is validating the `conversionMap` which is set by the developer.  The idea here then is that it fails early if the developer has passed in an invalid conversionMap.  Where I think we want to be a bit looser with regards to whether an error is thrown or not is in the context of the actual translated string.

Reply:
I’m confused because what it sounds like you are saying doesn’t match what the code says.

Did you mean to write `.some( e => ! isValid( e ) )`

The code currently says “if a developer puts at least one `WPElement` then we’ll assume the whole thing is valid”

It sounds like you are saying “we want to alert the developer at the first moment they pass a non-`WPElement`” which could be realized by either the `every` or an negated `some`

Reply:
oops ya, I missed the bang 🤦‍♂ I'll get that fixed.

Reply:
I decided to go with using `every` simply because it made the conditional a bit clearer.  I benchmarked too and while obviously some will exit sooner if there's an invalid element in a index at that beginning of the map, the variation between using `every` and `some` was inconsequential until the the map is much larger (>1000 length)  (which in the context of this use-case is unlikely it'll get very large).  See 6b7dfc2

Reply:
If we are able to discern that performance matters here then our best best is to skip `some` and `every` entirely. I think it's good to start here where the semantic is preferred over the implementation, but the `for` loop for some reason in JavaScript is almost always at least a few times faster [eg.](https://jsperf.com/array-some-vs-loop)

```js
const isValidConversionMap = ( conversionMap ) => {
	if ( 'object' !== typeof conversionMap ) {
		return;
	}

	const values = Object.values( conversionMap );
	for ( const i = 0; i < values.length; i++ ) {
		if ( ! isValidElement( values[ i ] ) ) {
			return false;
		}
	}

	return true;
}
```

But like I said, this is meant to be an informative comment, but until we have reason to believe that the iteration speed here (doing almost no processing over almost always less than five or six items) then we should be willing to let this stay more expressive in the code.

Alternatively, we could avoid this entirely and slip the validation into the parser itself. That would remove an entire iteration pass, allow for extra properties in the conversion map that we don't need and which may not be valid but which aren't causing harm, and not only enforce the structure but also enforce that we have valid components for exactly the names that we need.

Just more stewing.

Reply:
> Alternatively, we could avoid this entirely and slip the validation into the parser itself. That would remove an entire iteration pass, allow for extra properties in the conversion map that we don't need and which may not be valid but which aren't causing harm, and not only enforce the structure but also enforce that we have valid components for exactly the names that we need.

Yeaah, but then we're also parsing the translation string before we know if the conversion map is valid.  So I'm not sure there's significant gains here either way.  While this function _will_ be used, I'm not sure whether it will be a hot path being that the _majority_ of translation cases will probably be covered okay by existing `wp.i18n` functions.  Of course that might change when we make available the more expressive `<Translate>This is a <a>string</a></Translate>` syntax (in a separate pull) that could end up being used in more places than needed.

Reply:
> Yeaah, but then we're also parsing the translation string before we know if the conversion map is valid.

Practically I don't anticipate this mattering or being significant in any way. That parse should be blazing fast and it's not like it hurts anything to parse and fail. Either way the function call fails if the supplied components aren't right.